src/Statement/Grouper/FilteringStatementGrouper.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Statement\Grouper;
+
+use InvalidArgumentException;
+use Wikibase\DataModel\Statement\Statement;
+use Wikibase\DataModel\Statement\StatementFilter;
+use Wikibase\DataModel\Statement\StatementList;
+
+/**
+ * @since 3.2
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class FilteringStatementGrouper implements StatementGrouper {
```

Comment:
Why do we need statement groupers at all? Wouldn't it be sufficient to give the view code a filter?
E.g. instead of creating two groups for "ID statements" and "non-ID statements", just call the view code once with a filter that matches all ID statements, and once with a filter that matches all non-ID statements?

Also, I propose to change our internal code to work on per-property statement groups everywhere. We shouw statements grouped by property, and we group statements be property in the JSON representation. Wwe should probably also do this consistently in PHP. It's just needlessly painful to work on the flat statement list.